invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/components/requests.js

```diff
@@ -222,35 +242,75 @@ export const RDMEmptyResults = (props) => {
   );
 };
 
-export const RDMRequestsSearchLayout = (props) => (
-  <Container>
-    <Grid>
-      <Grid.Row columns={3}>
-        <Grid.Column width={4} />
-        <Grid.Column width={8}>
-          <SearchBar placeholder={i18next.t("Search requests...")} />
-        </Grid.Column>
-        <Grid.Column width={4}>
-          <Button
-            color="green"
-            icon="upload"
-            href="/requests/new"
-            content={i18next.t("New request")}
-            floated="right"
-          />
-        </Grid.Column>
-      </Grid.Row>
-      <Grid.Row>
-        <Grid.Column width={4}>
-          <SearchAppFacets aggs={props.config.aggs} />
-        </Grid.Column>
-        <Grid.Column width={12}>
-          <SearchAppResultsPane layoutOptions={props.config.layoutOptions} />
-        </Grid.Column>
-      </Grid.Row>
-    </Grid>
-  </Container>
-);
+export const RequestToggleComponent = ({
+  updateQueryFilters,
+  userSelectionFilters,
+}) => {
+  const [open, setOpen] = useState(true);
+
+  const retrieveOpenRequests = () => {
+    if (open) {
+      return;
+    }
+    setOpen(true);
+    const filterToRemove = userSelectionFilters.filter((obj1) => {
+      return obj1.includes("is_open");
+    });
+    const filters = filterToRemove ? filterToRemove : [];
+    filters.push(["is_open", "true"]);
+    updateQueryFilters(filters);
+  };
+
+  const retrieveClosedRequests = () => {
+    if (!open) {
+      return;
+    }
+    setOpen(false);
+    const filterToRemove = userSelectionFilters.filter((obj1) => {
+      return obj1.includes("is_open");
+    });
+    const filters = filterToRemove ? filterToRemove : [];
+    filters.push(["is_open", "false"]);
+    updateQueryFilters(filters);
+  };
+
+  return (
+    <Button.Group basic>
+      <Button onClick={retrieveOpenRequests} active={open}>
+        Open
+      </Button>
+      <Button onClick={retrieveClosedRequests} active={!open}>
+        Closed
+      </Button>
+    </Button.Group>
+  );
+};
+
+export const RDMRequestsSearchLayout = (props) => {
```

Comment:
I think now you can use the HOC I created for that: https://github.com/inveniosoftware/invenio-app-rdm/blob/feature/communities-ui/invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/user_dashboard/components/base.js#L113

Reply:
The layout is slightly different, the searchbar is not the same width and there is no newButton, also we have the open/closed filter, I don't think we can reuse it easily